src/utilities/findBreakingChanges.js

```diff
@@ -278,22 +354,128 @@ export function findFieldsThatChangedType(
           description: `${typeName}.${fieldName} was removed.`,
         });
       } else {
-        // Check if the field's type has changed in the new schema.
-        const oldFieldType = getNamedType(oldTypeFieldsDef[fieldName].type);
-        const newFieldType = getNamedType(newTypeFieldsDef[fieldName].type);
-        if (oldFieldType.name !== newFieldType.name) {
+        const oldFieldType = oldTypeFieldsDef[fieldName].type;
+        const newFieldType = newTypeFieldsDef[fieldName].type;
+
+        const isSafe =
+          isChangeSafeForInputObjectFieldOrFieldArg(oldFieldType, newFieldType);
+        if (!isSafe) {
+          const oldFieldTypeString = isNamedType(oldFieldType) ?
+            oldFieldType.name :
+            oldFieldType.toString();
+          const newFieldTypeString = isNamedType(newFieldType) ?
+            newFieldType.name :
+            newFieldType.toString();
           breakingFieldChanges.push({
             type: BreakingChangeType.FIELD_CHANGED_KIND,
             description: `${typeName}.${fieldName} changed type from ` +
-              `${oldFieldType.name} to ${newFieldType.name}.`,
+              `${oldFieldTypeString} to ${newFieldTypeString}.`,
           });
         }
       }
     });
+    // Check if a non-null field was added to the input object type
+    Object.keys(newTypeFieldsDef).forEach(fieldName => {
+      if (
+        !(fieldName in oldTypeFieldsDef) &&
+        newTypeFieldsDef[fieldName].type instanceof GraphQLNonNull
+      ) {
+        breakingFieldChanges.push({
+          type: BreakingChangeType.NON_NULL_INPUT_FIELD_ADDED,
+          description: `A non-null field ${fieldName} on ` +
+            `input type ${newType.name} was added.`,
+        });
+      }
+    });
   });
   return breakingFieldChanges;
 }
 
+function isChangeSafeForObjectOrInterfaceField(
+  oldType: GraphQLType,
+  newType: GraphQLType,
+): boolean {
```

Comment:
The `isSafe` var is leftover from before these were factored out into separate functions. Updated so that we just have inline returns.